### PR TITLE
CWL tool wrappers for bwa, bedtools, samtools

### PR DESCRIPTION
**Pull request by portah**
This just a first step to describe bedtools genomecov [OPTIONS] -i "file" -g "genome file"
1. If input file in one of the bed/gff/vcf formats then prefix is -"i" if it is .bam the prefix is "-ibam"
2. Genome file must be in "chromName""tab""chromSize" format
3. For the beginning options has to be optional

### CODE REVIEW

**Comment by portah** [comment]
I did an update bedtools-genomecov requires two params input bam file and genome file. All other params are optional. I don't know how to check type of input file (based on extension) and change prefix -i or -ibam. And force to use one of the params if the other is chosen (like --scale requires either -bg or -bga). The only default engine is JsonPointer or I can use JS?

**Comment by mr-c** [comment]
I think I'll use this PR to setup an autotester using travis. Are these tests resource intensive?

**Comment by portah** [comment]
No they are not. All of them one threaded and can be done on a laptop or VM

File: docker/bedtools2/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM ubuntu:15.04
+MAINTAINER Andrey V Kartashov "porter@porter.st"
+COPY bedtools /usr/local/bin/bedtools
```
Comment (mr-c) [inline docker/bedtools2/Dockerfile]: Can the version of bedtools in Ubuntu be used instead? http://packages.ubuntu.com/vivid/bedtools

File: workflows/scidap/bam-genomecov-bigwig-job.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+    "scidap_working_folder": "/Users/porter/Work/scidap/workflows/workflows/scidap/",
```
Comment (mr-c) [inline workflows/scidap/bam-genomecov-bigwig-job.json]: I'm surprised that this path works; is there a more portable way?

Comment (portah) [inline workflows/scidap/bam-genomecov-bigwig-job.json]: It is not an input for CWL. I'll remove it

File: docker/STAR/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,3 @@
+FROM ubuntu:15.04
+MAINTAINER Andrey V Kartashov "porter@porter.st"
+COPY STAR /usr/local/bin/STAR
```
Comment (mr-c) [inline docker/STAR/Dockerfile]: This is also available in Ubuntu: http://packages.ubuntu.com/wily/rna-star

Comment (portah) [inline docker/STAR/Dockerfile]: Some of the tools a re there, some of the tools can not be there like the latest samtools which is not yet released but latest github revision has important bug fixes

Comment (mr-c) [inline docker/STAR/Dockerfile]: Indeed! The pre-packaged software is a preference, not a requirement. Gotta use what is available to get the job done :-)

**Comment by FarahZKhan** [comment]
Hello, below is the list of GATK tools I was thinking to use for the workflow: 
•	Mapping: Map with BWA for Illumina
•	Mark duplicates: Picard  MarkDuplicates
•	Indel Realignment: RealignerTargetCreator and IndelRealigner
•	Base Recalibration: BaseRecalibrator
•	Data compression with Reduce Reads: ReduceReads
•	Variant Calling: HaplotypeCaller
•	Variant recalibration: VariantRecalibrator and ApplyRecalibration
•	Variant Evaluation: VariantEval
I am assuming none of the tools are implemented using CWL as yet. What would be the best place to start? Thank you for all the help earlier,

**Pull request merged**